From 9b4ae951eeabc09fda80e446434ef8f03849522e Mon Sep 17 00:00:00 2001
From: Roy Pledge <roy.pledge@nxp.com>
Date: Fri, 22 Jun 2018 15:47:07 -0400
Subject: [PATCH] staging/fsl_qbman: Consume all frames in DQRR during init

The qm_dqrr_cdc_consume_n() function takes a bitmask, not an index.

Signed-off-by: Roy Pledge <roy.pledge@nxp.com>
---
 drivers/staging/fsl_qbman/qman_low.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/staging/fsl_qbman/qman_low.h
+++ b/drivers/staging/fsl_qbman/qman_low.h
@@ -658,7 +658,7 @@ static inline int qm_dqrr_init(struct qm
 			qm_dqrr_cce_consume(portal, dqrr->fill);
 			break;
 		case qm_dqrr_cdc:
-			qm_dqrr_cdc_consume_n(portal, (QM_DQRR_SIZE - 1));
+			qm_dqrr_cdc_consume_n(portal, (1<<QM_DQRR_SIZE) - 1);
 			break;
 		default:
 			DPA_ASSERT(0);
